src/plugins/saved_objects_management/common/types/v1.ts

```diff
@@ -0,0 +1,191 @@
+ * which is intended for server-side only use.
+ */
+export interface SavedObjectWithMetadataV1<T = unknown> {
+  id: string;
+  type: string;
+  meta: SavedObjectMetadataV1;
+  error?: SavedObjectError;
+  created_at?: string;
+  updated_at?: string;
+  attributes: T;
+  namespaces?: string[];
+  references: SavedObjectReferenceV1[];
+}
+
+export type SavedObjectRelationKindV1 = 'child' | 'parent';
+
+/**
+ * {@inheritdoc SavedObjectRelation}
+ */
+export interface SavedObjectRelationV1 {
+  id: string;
+  type: string;
+  relationship: SavedObjectRelationKindV1;
+  meta: SavedObjectMetadataV1;
+}
+
+/**
+ * Represents a relation between two {@link SavedObject | saved object}
+ */
+export type SavedObjectRelation = SavedObjectRelationV1;
+
+/**
+ * {@inheritdoc SavedObjectInvalidRelation}
+ */
+export interface SavedObjectInvalidRelationV1 {
+  id: string;
+  type: string;
+  relationship: SavedObjectRelationKindV1;
+  error: string;
+}
+
+/**
+ * Represents a relation between two {@link SavedObject | saved object}
+ */
+export type SavedObjectInvalidRelation = SavedObjectInvalidRelationV1;
+
+export type SavedObjectGetRelationshipsResponseV1 = RelationshipsResponseHTTPV1;
+
+export interface SavedObjectManagementTypeInfoV1 {
+  name: string;
+  // TODO: Fix. We should not directly expose these values to public code.
+  namespaceType: SavedObjectsNamespaceType;
+  hidden: boolean;
+  displayName: string;
+}
+export type SavedObjectManagementTypeInfo = SavedObjectManagementTypeInfoV1;
+
+/** HTTP API interfaces */
+
+export type BulkGetHTTPBodyV1 = Array<{
+  id: string;
+  type: string;
+}>;
+
+export type BulkGetHTTPResponseV1 = SavedObjectWithMetadataV1[];
+
+export type BulkDeleteHTTPBodyV1 = Array<{
+  type: string;
+  id: string;
+}>;
+
+export type BulkDeleteHTTPResponseV1 = Array<{
+  /** The ID of the saved object */
+  id: string;
+  /** The type of the saved object */
+  type: string;
+  /** The status of deleting the object: true for deleted, false for error */
+  success: boolean;
+  /** Reason the object could not be deleted (success is false) */
+  error?: SavedObjectError;
+}>;
+
+export type FindSearchOperatorHTTPV1 = 'AND' | 'OR';
+export type FindSortOrderHTTPV1 = 'asc' | 'desc';
+
+export interface ReferenceHTTPV1 {
+  type: string;
+  id: string;
+}
+
+export interface FindQueryHTTPV1 {
+  perPage?: number;
+  page?: number;
+  type: string | string[];
+  // TODO: Fix. this API allows writing an arbitrary query that is passed straight to our persistence layer, thus leaking SO attributes to the public...
+  search?: string;
+  defaultSearchOperator?: FindSearchOperatorHTTPV1;
+  // TODO: Fix. this API allows sorting by any field, thus leaking SO attributes to the public...
+  sortField?: string;
+  sortOrder?: FindSortOrderHTTPV1;
+  hasReference?: ReferenceHTTPV1 | ReferenceHTTPV1[];
+  hasReferenceOperator?: FindSearchOperatorHTTPV1;
+  // TODO: Fix. This exposes attribute schemas to clients.
```

Comment:
Comments like this represent tech debt that, to be fully compatible with versionable APIs should be reviewed.

Reply:
Some of those comments are relatively old. I agree though, we should resolve as much tech debt as possible now.